components/build-templates/base/e2e/serviceaccount.yaml

```diff
@@ -2,9 +2,9 @@ apiVersion: v1
 kind: ServiceAccount
 metadata:
   name: appstudio-pipeline
-  namespace: build-templates-e2e
```

Comment:
I think this is needed, otherwise it will be created in the build-templates namspace

Also there was a reason why we did not keep the service account in base, but duplicated it for production and stage. The reason is that it the SA file will then be deployed to dev env as well. But it will fail to find the secrets, causing problems.